### ISSUE TITLE
code-server serve-local with --host switch doesn't work well on hosts with multiple IPs

### ISSUE THREAD

**Issue by marmolak** [microsoft/vscode-remote-release#7202]
- VSCode Version: code-1.71.2-1663191299.el7.x86_64
- Local OS Version: Fedora 36
- Remote OS Version: Fedora 36
- Remote Extension/Connection Type: Server
- Logs: Provided after reproducer

There is not intuitive behaviour when host with multiple A records were used as a --host argument.

Steps to Reproduce:

1. install code-server from:
https://code.visualstudio.com/blogs/2022/07/07/vscode-server
via:
```
wget -O- https://aka.ms/install-vscode-server/setup.sh | sh
```
2. Try to start with:
```
code-server serve-local --disable-telemetry --host bigoook.local
```

Result:
```
Server bound to 172.17.0.1:8000 (IPv4)
Extension host agent listening on 8000

Web UI available at http://localhost:8000/?tkn=ca69db51-223f-4447-8f0c-f70d1d984678
[14:57:25] Extension host agent started.
```

bigoook.local is name provided by avahi mDNS:

```
$ dig bigoook.local
...
bigoook.local.          0       IN      A       192.168.32.101
bigoook.local.          0       IN      A       192.168.200.9
bigoook.local.          0       IN      A       172.18.0.1
bigoook.local.          0       IN      A       172.17.0.1
...
```

Does this issue occur when you try this locally?: Yes
Does this issue occur when you try this locally and all extensions are disabled?: Yes

**Comment by mat-wang** [comment]
How about using --host 0.0.0.0?

**Comment by marmolak** [comment]
> How about using --host 0.0.0.0?

That works:
```
Server bound to 0.0.0.0:8000 (IPv4)
Extension host agent listening on 8000
                                                                                                              Web UI available at http://localhost:8000/?tkn=ca69db51-xxxx-xxx7-8f0c-f70d1d984678
[14:30:15] Extension host agent started.
```

however, hint url is still wrong.

**Comment by mat-wang** [comment]
> > How about using --host 0.0.0.0?
> 
> 
> 
> That works:
> 
> ```
> 
> Server bound to 0.0.0.0:8000 (IPv4)
> 
> Extension host agent listening on 8000
> 
>                                                                                                               Web UI available at http://localhost:8000/?tkn=ca69db51-xxxx-xxx7-8f0c-f70d1d984678
> 
> [14:30:15] Extension host agent started.
> 
> ```
> 
> 
> 
> however, hint url is still wrong.

What's your meaning for hint url? 

The 'host' option specify where the connection comes from. For example, if it is set to localhost, the server will only response the request from localhost and refuse all request from other request. Here set to 0.0.0.0 means it will accept the response from all ip addresses. But the server will only listen on your local machine.

**Comment by marmolak** [comment]
Hello. I understand usage of 0.0.0.0, I just reported something, which can other users can find.

Seems like formating of previous comment is broken. Sorry for that.
 By hint I mean string:

`Web UI available at http://localhost:8000/?tkn=ca69db51-xxxx-xxx7-8f0c-f70d1d984678`

Which is not true in case that you run vs code server on remote machine where 0.0.0.0 (or other public IP) is used as a host switch parameter.

**Comment by aeschli** [comment]
code-server as been superceeded by VS Code tunnels. This is not a scenario we support yet.
